x/pairing/keeper/unresponsive_provider.go

```diff
@@ -111,84 +109,118 @@ func (k Keeper) getUnresponsiveProvidersToPunish(ctx sdk.Context, currentEpoch u
 
 // Function to count the CU serviced by the unresponsive provider or the CU of the complainers.
 // The function counts CU from <epoch> back. The number of epochs to go back is <epochsNumToGoBack>. The function returns the last visited epoch
-func (k Keeper) countCuForUnresponsiveProviderAndComplainers(ctx sdk.Context, isProvider bool, epoch uint64, epochsNumToGoBack uint64, providersPunishEntryMap map[string]punishEntry) (uint64, error) {
+func (k Keeper) updateCuForProviderInProviderCuCounterForUnreponsivenessMap(ctx sdk.Context, epoch uint64, epochsNumToCheckCUForUnresponsiveProvider uint64, epochsNumToCheckCUForComplainers uint64, providerStakeEntry epochstoragetypes.StakeEntry, providerCuCounterForUnreponsivenessMap map[string]providerCuCounterForUnreponsiveness) error {
 	epochTemp := epoch
+	createNewproviderCuCounterForUnreponsivenessEntryFlag := false
+	// get the provider's SDK account address
+	sdkStakeEntryProviderAddress, err := sdk.AccAddressFromBech32(providerStakeEntry.GetAddress())
+	if err != nil {
+		return utils.LavaFormatError("unable to sdk.AccAddressFromBech32(provider)", err, &map[string]string{"provider_address": providerStakeEntry.Address})
+	}
 
-	for counter := uint64(0); counter < epochsNumToGoBack; counter++ {
-		// Get previous epoch (from epochTemp)
-		previousEpoch, err := k.epochStorageKeeper.GetPreviousEpochStartForBlock(ctx, epochTemp)
-		if err != nil {
-			return 0, utils.LavaError(ctx, k.Logger(ctx), "get_previous_epoch", map[string]string{"err": err.Error(), "epoch": fmt.Sprintf("%+v", epochTemp)}, "couldn't get previous epoch")
-		}
+	// get the provider's providerCuCounterForUnreponsiveness
+	providerCuCounterForUnreponsivenessEntry, found, providerCuCounterForUnreponsivenessKey := k.getProviderCuCounterForUnreponsivenessEntryFromMap(providerStakeEntry, providerCuCounterForUnreponsivenessMap)
+	if !found {
+		// if providerCuCounterForUnreponsivenessEntry is not found, raise a flag to create new entry in providerCuCounterForUnreponsivenessMap
+		createNewproviderCuCounterForUnreponsivenessEntryFlag = true
```

Comment:
instead of creating a flag just create an empty entry here

Reply:
done